core/io/resource_uid.cpp

```diff
@@ -346,10 +349,26 @@ String ResourceUID::get_path_from_cache(Ref<FileAccess> &p_cache_file, const Str
 	return String();
 }
 
-void ResourceUID::clear() {
-	cache_entries = 0;
-	unique_ids.clear();
-	changed = false;
+void ResourceUID::verify() {
+	MutexLock l(mutex);
+
+	Vector<ID> invalid_ids;
+	Ref<FileAccess> f = FileAccess::create(FileAccess::ACCESS_RESOURCES);
+
+	for (KeyValue<ID, Cache> &E : unique_ids) {
+		const String file = String::utf8(E.value.cs.ptr());
+		if (!f->file_exists(file) || ResourceLoader::get_resource_uid(file) != E.key) {
```

Comment:
Invoking ResourceLoader for every file is very costly.
I checked with my big project and it seems to take considerably long time on startup. This should be only performed if the file could've changed since last scan (based on modified time most likely).

Reply:
Indeed. However, it's currently difficult to implement before the first scan.

I'm going to make the `EditorFileSystem` and `UID` cache files more reliable and rely on those for checks.

First, build the file system tree based on the cache files. Then, check for file changes using the scan_changes logic.

There are some commits currently, but they're not complete yet: https://github.com/Rindbee/godot/tree/make-editor-file-system-more-reliable